core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -814,7 +833,10 @@ class Partition(val topicPartition: TopicPartition,
   private def tryCompleteDelayedRequests(): Unit = delayedOperations.checkAndCompleteAll()
 
   def maybeShrinkIsr(): Unit = {
-    val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
+    val needsIsrUpdate = inReadLock(leaderIsrUpdateLock) {
```

Comment:
Not a major issue, but since `needsShrinkIsr` is exposed, maybe we can move the lock acquire to it?

Reply:
Actually `needsShrinkIsr` can be private, updated. Left the lock in here to be consistent with `needsExpandIsr`.